proto/dataplane/dataplane.proto

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package lemming.dataplane;
+
+option go_package = "github.com/openconfig/lemming/proto/dataplane";
+
+message UpdatePortRequest {
+  string name = 1;
+  string hwaddr = 2;
+  repeated IPNetwork ipv4 = 3;
+  repeated IPNetwork ipv6 = 4;
+}
+
+message UpdatePortResponse {}
+
+enum PortState {
+  PORTSTATE_INVALID = 0;
+  PORTSTATE_DOWN = 1;
+  PORTSTATE_UP = 2;
+}
+
+message SetPortStateRequest {
+  string name = 1;
+  PortState admin_state = 2;
+}
+
+message SetPortStateResponse {}
+
+message SubscribePortStateRequest {}
+
+message IPNetwork {
+  string ip = 1;
+  uint64 prefix_len = 3;
+}
+
+message SubscribePortStateResponse {
+  string name = 1;
+  PortState admin_state = 2;
+  repeated IPNetwork network = 3;
+}
+
+message RouteDst {
+  string port = 1;
+  string ip = 2;
+  uint64 weight = 3;
+}
+
+message InsertRouteRequest {
+  uint64 vrf = 1;
+  string prefix = 2;
+  repeated RouteDst dst = 3;
+}
+
+message InsertRouteResponse {}
+
+message DeleteRouteRequest {
```

Comment:
For modifying a route is the expectation that one calls delete and then add, or does add for an existing route replace one that is already there?

Reply:
my idea was to call delete then add, but i'm don't have a strong opinion

Reply:
I think each route should be uniquely identified by a key, namely (prefix, vrf, address-family) as fields I think is enough. If so, then I would prefer "add" to mean a replace, since that sounds to me like the simplest semantic to give this operation in the presence of an existing route.

Reply:
updated comment to clarify behavior and allow overwriting